### ISSUE TITLE
Sensible return-value from job_queue_node_update_status_simple()

### ISSUE THREAD

**Issue by BjarneHerland** [equinor/ert#3323]
The function `job_queue_node_update_status_simple()` seems to always return `false` and might be changed to a void-function.  (Probably left-over from cut'n'paste from `job_queue_node_update_status()`)

**Comment by valentin-krasontovitsch** [comment]
Now I've started to look at the code a bit, following your suggestion, and I've got some questions : )

The first one I would already like to formulate. There's quite a bit of code to read here, I feel, and I'm trying to get a feeling for what's happening, what the objects are, and how they are connected. Naming is an important guide for me here, so I have to ask about a perceived inconsistency - take the two files in `libres/lib/job_queue`, `job_queue_status.cpp` and `job_status.cpp` - they each refer to the same list of states, each doing slighty different things with the list (job_status.cpp just translates numbers representing states to strings describing the states).

I have to wonder - is there a difference between a job status and and a job queue status, or are those two synonyms?

I also see that `libres/lib/include/ert/job_queue/job_status.hpp` and `libres/lib/include/ert/job_queue/job_queue_status.hpp` indeed seem to define two different types, the former being an enum type `job_status_type`, the other a struct `job_queue_status_type`.

Another question - there exists `job_queue_status_transition` and `job_queue_node_status_transition`. There seem to be some entities at play here, job, queue, and node. What is the relationship between those, and which one of them has a status that we are interested in?

If you prefer to do this in person rather then in writing, I will be at the office tomorrow.

**Comment by valentin-krasontovitsch** [comment]
After a bit of live discussion, we indeed have two states - a job status in memort and a job status on the node, that needs to be queried / updated using the `driver`. besides this conceptuality, we decided to try to not delve too deeply into the code and just try to fix the issue with what we know and understand.

looking at the code more carefully and considering where the function whose return value (and type) we are modifying is used, it becomes rather clear what changes are needed - see the connected PR.
